File: SmartHome/app/ingternal/automation/schemas/automation.py
from enum import Enum
from pydantic import BaseModel
from .enums import ConditionType, Operation, SetType
from typing import Optional, List, Dict

class TriggerItemSchema(BaseModel):
	service: str
	object: str
	data: str

class ConditionItemSchema(BaseModel):
	operation: Operation
	arg1_service: str
	arg1_object: str
	arg1_data: str
	arg2_service: str
	arg2_object: str
	arg2_data: str
	
	class Config:  
		use_enum_values = True

class ActionItemSchema(BaseModel):
	service: str
	object: str
	field: str
	data: str
	type_set: SetType

	class Config:  
		use_enum_values = True

class AutomationSchema(BaseModel):
	name:str
	trigger:List[TriggerItemSchema]
	condition: List[ConditionItemSchema]
	condition_type: ConditionType
	then: List[ActionItemSchema]
	else_branch: List[ActionItemSchema]
	is_enabled: bool = True
	

	class Config:  
		use_enum_values = True


class EnableSchema(BaseModel):
	is_enabled: bool = True
	

File: frontend/client_base/src/entites/automation/models/automation.ts
// Enums corresponding to ConditionType, Operation, and SetType
export enum ConditionType {
    AND = "and",
    OR = "or"
}

export enum Operation {
    MORE = ">",
    LESS = "<",
    MORE_OR_EQUAL = ">=",
    LESS_OR_EQUAL = "<=",
    EQUAL = "==",
    NOT_EQUAL = "!="
}

export enum SetType {
    DATA = "data",
    COMMAND = "command"
}

// Trigger item interface
export interface TriggerItem {
    service: string;
    object: string;
    data: string;
}

// Condition item interface
export interface ConditionItem {
    operation: Operation;
    arg1_service: string;
    arg1_object: string;
    arg1_data: string;
    arg2_service: string;
    arg2_object: string;
    arg2_data: string;
}

// Action item interface
export interface ActionItem {
    service: string;
    object: string;
    field: string;
    data: string;
    type_set: SetType;
}

// Automation schema interface
export interface Automation {
    name: string;
    trigger: TriggerItem[];
    condition: ConditionItem[];
    condition_type: ConditionType;
    then: ActionItem[];
    else_branch: ActionItem[];
    is_enabled?: boolean; // Optional, defaults to true in Python
}

// Enable schema interface
export interface EnableSchema {
    is_enabled?: boolean; // Optional, defaults to true in Python
}
